src/MxpTag.cpp

```diff
@@ -0,0 +1,126 @@
+/***************************************************************************
+ *   Copyright (C) 2020 by Gustavo Sousa - gustavocms@gmail.com            *
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ *   This program is distributed in the hope that it will be useful,       *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU General Public License for more details.                          *
+ *                                                                         *
+ *   You should have received a copy of the GNU General Public License     *
+ *   along with this program; if not, write to the                         *
+ *   Free Software Foundation, Inc.,                                       *
+ *   59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.             *
+ ***************************************************************************/
+
+#include "MxpTag.h"
+#include "TMxpTagParser.h"
+
+const MxpTagAttribute& MxpStartTag::getAttribute(int attrIndex) const
+{
+    return getAttribute(mAttrsNames[attrIndex]);
+}
+
+const MxpTagAttribute& MxpStartTag::getAttribute(const QString& attrName) const
+{
+    const auto ptr = mAttrsMap.find(attrName.toUpper());
+    return *ptr;
```

Comment:
Strange - i thought I should have spotted that '&' - guess I've been staring at too much code. :dark_sunglasses:

Reply:
Ah, hang on - you **are** returning a reference, and it refers to, let me think, a result from using `QMap<T1, T2>::find(const T1 &)` which returns an iterator pointing to the T2 type value with the given key. Maybe that *is* safe after all. {I've heard nasty tales of returning references to variables with local scope but that doesn't seem to be the case here, right?} :confounded:

Reply:
> Ah, hang on - you **are** returning a reference, and it refers to, let me think, a result from using `QMap<T1, T2>::find(const T1 &)` which returns an iterator pointing to the T2 type value with the given key. Maybe that _is_ safe after all. {I've heard nasty tales of returning references to variables with local scope but that doesn't seem to be the case here, right?}

Exactly, this is to avoid returning a local reference. But I think the class can be redesigned to hold a value instead of reference.